compiler/qsc_linter/src/tests.rs

```diff
@@ -14,6 +14,32 @@ use qsc_frontend::compile::{self, CompileUnit, PackageStore, SourceMap};
 use qsc_hir::hir::CallableKind;
 use qsc_passes::PackageType;
 
+#[test]
+fn set_keyword_lint() {
+    check(
+        &wrap_in_callable("set x = 3;", CallableKind::Function),
+        &expect![[r#"
+            [
+                SrcLint {
+                    source: "set",
+                    level: Allow,
+                    message: "deprecated use of `set` keyword",
+                    help: "`set` keywords are deprecated for assignments, they can be removed",
```

Comment:
nitpick: I think I prefer this phrasing
```suggestion
                    help: "`set` keywords are deprecated for assignments and can be removed",
```